content/en/blog/_posts/2022-11-11-retroactive-default-storage-class.md

```diff
@@ -0,0 +1,163 @@
+Let's take a closer look at each of them.
+
+### Changing default StorageClass did not have a good solution
+
+With the alpha feature enabled, there were two options admins had when they wanted to change the default StorageClass:
+
+1. Creating a new StorageClass as default before removing the old one associated with the PVC. 
+This would result in having two defaults for a short period.
+At this point, if a user were to create a PersistentVolumeClaim with storageClassName set to <code>null</code> (implying default StorageClass), the newest default StorageClass would be chosen and assigned to this PVC.
+
+2. Removing the old default first and creating a new default StorageClass.
+This would result in having no default for a short time.
+Subsequently, if a user were to create a PersistentVolumeClaim with storageClassName set to <code>null</code> (implying default StorageClass), the PVC would be in <code>Pending</code> state forever.
+The user would have to fix this by deleting the PVC and recreating it once the default StorageClass was available.
+
+
+### Resource ordering during cluster installation
+
+If a cluster installation tool needed to create resources that required storage, e.g., image registry, it was difficult to get the ordering right. 
+This is because any Pods that required storage would rely on the presence of a default StorageClass and would fail to be created if it wasn't defined.
+
+## What changed
+
+We've changed the PersistentVolume (PV) controller to assign a default StorageClass to any unbound PersistentVolumeClaim that has the storageClassName set to <code>null</code>.
+We've also modified the PersistentVolumeClaim admission within the API server to allow the change of values from unset value to an actual StorageClass name.
+
+### <code>nil</code> vs <code>""</code> for storageClassName - does it matter?
+
+Before this feature was introduced, those values were equal in terms of behavior. Any PersistentVolumeClaim with the storageClassName set to <code>nil</code> or <code>""</code> would bind to an existing PersistentVolume resource with storageClassName also set to <code>nil</code> or <code>""</code>.
+
+With this new feature enabled we wanted to maintain this behavior but also be able to update the StorageClass name.
+With these constraints in mind, the feature changes the semantics of <code>nil</code>. If a default StorageClass is present, <code>nil<code> would translate to "Give me a default" and <code>""</code> would mean "Give me PersistentVolume that also has <code>""</code> StorageClass name." In the absence of a StorageClass, the behavior would remain unchanged.
+
+Summarizing the above, we've changed the semantics of <code>nil<code> so that its behavior depends on the presence or absence of a definition of default StorageClass.
+
+The tables below show all these cases to better describe when PVC binds and when it's StorageClass gets updated.
+
+<table>
```

Comment:
I'd use a single “=“ if not writing it inside `<code>...</code>`; also, I'd put `storageClassName` in monospace (eg `<tt>storageClassName</tt>`).

Reply:
Done.